internal/v5/resources.go

```diff
@@ -132,7 +134,14 @@ func (h *ReqHandler) serveDownloadResourceDocker(id *router.ResolvedURL, r *mong
 		}
 		resp.Password = password
 	}
-	return httprequest.WriteJSON(w, http.StatusOK, resp)
+	buf, err := json.Marshal(resp)
+	if err != nil {
+		return errgo.Mask(err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set(params.ContentHashHeader, fmt.Sprintf("%x", sha512.Sum384(buf)))
+	_, err = w.Write(buf)
+	return errgo.Mask(err)
```

Comment:
Please don't return an error if the write fails. There's nothing we can do in that case other than produce an inconsistent body.